node/src/main/scala/com/wavesplatform/Application.scala

```diff
@@ -143,6 +153,32 @@ class Application(val actorSystem: ActorSystem, val settings: WavesSettings, con
       }(scheduler)
 
     val historyReplier = new HistoryReplier(blockchainUpdater, settings.synchronizationSettings, historyRepliesScheduler)
+
+    // Extensions start
+    val extensionContext = new Context {
+      override def settings: WavesSettings                                  = app.settings
+      override def blockchain: Blockchain with BlockchainUpdater            = app.blockchainUpdater
+      override def time: Time                                               = app.time
+      override def wallet: Wallet                                           = app.wallet
+      override def utx: UtxPool                                             = utxStorage
+      override def broadcastTx(tx: Transaction): Unit                       = allChannels.broadcastTx(tx, None)
+      override def spendableBalanceChanged: Observable[(Address, Asset)]    = app.spendableBalanceChanged
+      override def actorSystem: ActorSystem                                 = app.actorSystem
+      override def blockchainUpdated: Option[Observable[BlockchainUpdated]] = maybeBlockchainUpdated
```

Comment:
```suggestion
      override def blockchainUpdated: Observable[BlockchainUpdated] = ???
```
When update notifications are disabled, this could just be a dummy observable